BLSQ/iaso
# IA-3491, IA-3489: deploy doc on gh pages, change theme to material

Related JIRA tickets : IA-3491, IA-3489

## Self proofreading checklist


## Doc
comments in mkdocs.yml

## Changes

- Change theme to "material" in mkdocs.yml
- Add extra css to match theme primary color with iaso prod
- Add github action to build and deploy doc on github pages when pushing on main

## How to test

- Go to pages.openiaso.com. The current deployed theme should be "readthedocs" (see picture)

![Screenshot 2024-10-09 at 15 02 00](https://github.com/user-attachments/assets/15370b3e-c5da-483f-a3ae-ae0401e54e90)

- In `deploy_doc.yml`, replace `main` on line 13 with `IA-3491_deploy_doc_on_gh_pages`. Commit and push the change

--> You should see the action in the Github actions panel

Once the action is done running, go back to pages.openiaso.com: the theme should now be material.

IMPORTANT: in `deploy_doc.yml`,  restore the initial value (`main`) on line 13

Print screen / video
Material theme:

![Screenshot 2024-10-09 at 15 06 26](https://github.com/user-attachments/assets/529cccaf-6898-4237-bdec-d90ab9319192)

## Notes

Once the PR has been merged, we still need to change the CNAME to point to docs.openiaso.com, and cancel the existing (obsolete) record.
